src/reducer.ts

```diff
@@ -50,37 +41,49 @@ const { reducer, actions } = createSlice({
       state.currentSelection = null
     },
     removeMarker(state, action: PayloadAction<Marker>) {
-      state.markers = state.markers.filter(
-        (marker) => marker.id !== action.payload.id,
-      )
-      state.arrows = removeArrowsWithDependency(state.arrows, action.payload.id)
+      delete state.markers[action.payload.id]
+      // TODO: mutating record while iterating over it
```

Comment:
:scream:

Reply:
I missed this because I usually mark these as `TODAN`! thanks for catching it